src/adapters/refract/transformResource.coffee

```diff
@@ -0,0 +1,122 @@
+    })
+
+    requests = []
+    responses = []
+
+    _.httpTransactions(transitionElement).forEach((httpTransaction) ->
+      httpRequest = _(httpTransaction).httpRequests().first()
+      httpResponse  = _(httpTransaction).httpResponses().first()
+
+      httpRequestBody = _(httpRequest).messageBodies().first()
+      httpResponseBody = _(httpResponse).messageBodies().first()
+
+      httpRequestBodySchemas = _(httpRequest).messageBodySchemas().first()
+      httpResponseBodySchemas = _(httpResponse).messageBodySchemas().first()
+
+      httpRequestDescription = getDescription(httpRequest)
+      httpResponseDescription = getDescription(httpResponse)
+
+      requestAttributes = _.dataStructures(httpRequestBody)
+      requestAttributes = undefined if _.isEmpty(requestAttributes)
+
+      responseAttributes = _.dataStructures(httpResponseBody)
+      responseAttributes = undefined if _.isEmpty(responseAttributes)
+
+      # In refract just here we have method and href
+      resource.method = _.chain(httpRequest).get('attributes.method', '').contentOrValue().value()
+      resource.actionUriTemplate = _.chain(httpRequest).get('attributes.href', '').contentOrValue().value()
+
+      requestParameters = getUriParameters(_.get(httpRequest, 'attributes.hrefVariables'))
+      actionParameters = actionParameters.concat(requestParameters)
+
+      request = new blueprintApi.Request({
+        name: _.chain(httpRequest).get('meta.title', '').contentOrValue().value()
+        description: httpRequestDescription.raw
+        htmlDescription: httpRequestDescription.html
+        headers: getHeaders(httpRequest)
+        # reference
+        body: if _.content(httpRequestBody) then _.content(httpRequestBody) else ''
+        schema: if _.content(httpRequestBodySchemas) then _.content(httpRequestBodySchemas) else ''
+        # exampleId
+        attributes: requestAttributes
+        resolvedAttributes: requestAttributes
+      })
+
+      response = new blueprintApi.Response({
+        status: _.chain(httpResponse).get('attributes.statusCode').contentOrValue().value()
+        description: httpResponseDescription.raw
+        htmlDescription: httpResponseDescription.html
+        headers: getHeaders(httpResponse)
+        # reference
+        body: if _.content(httpResponseBody) then _.content(httpResponseBody) else ''
+        schema: if _.content(httpResponseBodySchemas) then _.content(httpResponseBodySchemas) else ''
+        # exampleId
+        attributes: responseAttributes
+        resolvedAttributes: responseAttributes
+      })
+
+      requests.push(request)
+      responses.push(response)
+    )
+
+    resource.requests = requests
+    resource.request = requests[0]
+    resource.responses = responses
+
+    resource.resourceParameters = resourceParameters
+    resource.actionParameters = actionParameters
+    resource.parameters = resourceParameters.concat(actionParameters)
+
+    resources.push(resource)
+  )
+
+  resources
```

Comment:
Can I safely assume that this is just a copy/paste of the original code? Is there any logic changed here?